### PR TITLE
[BEAM-7530] Add it test to read None values from BigQuery

### PR DESCRIPTION
**Pull request by Juta**
Post-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

Lang | SDK | Apex | Dataflow | Flink | Gearpump | Samza | Spark
--- | --- | --- | --- | --- | --- | --- | ---
Go | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Flink/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Go_VR_Spark/lastCompletedBuild/)
Java | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Apex/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Dataflow/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Flink/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Batch/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Flink_Streaming/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Gearpump/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Samza/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PostCommit_Java_ValidatesRunner_Spark/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Java_PVR_Spark_Batch/lastCompletedBuild/)
Python | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_Verify/lastCompletedBuild/)<br>[Build Status](https://builds.apache.org/job/beam_PostCommit_Python3_Verify/lastCompletedBuild/) | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_VR_Dataflow/lastCompletedBuild/) <br> [Build Status](https://builds.apache.org/job/beam_PostCommit_Py_ValCont/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_PVR_Flink_Cron/lastCompletedBuild/) | --- | --- | [Build Status](https://builds.apache.org/job/beam_PostCommit_Python_VR_Spark/lastCompletedBuild/)

Pre-Commit Tests Status (on master branch)
------------------------------------------------------------------------------------------------

--- |Java | Python | Go | Website
--- | --- | --- | --- | ---
Non-portable | [Build Status](https://builds.apache.org/job/beam_PreCommit_Java_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Python_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Go_Cron/lastCompletedBuild/) | [Build Status](https://builds.apache.org/job/beam_PreCommit_Website_Cron/lastCompletedBuild/) 
Portable | --- | [Build Status](https://builds.apache.org/job/beam_PreCommit_Portable_Python_Cron/lastCompletedBuild/) | --- | ---

See [.test-infra/jenkins/README](https://github.com/apache/beam/blob/master/.test-infra/jenkins/README.md) for trigger phrase, status and link of all Jenkins jobs.

### CODE REVIEW

**Comment by Juta** [comment]
Run Python PostCommit

**Comment by Juta** [comment]
Run Python PostCommit

**Comment by Juta** [comment]
Run Python PostCommit

**Comment by Juta** [comment]
R: @tvalentyn

File: sdks/python/apache_beam/io/gcp/bigquery_read_it_test.py [hunk 1]
```diff
@@ -172,6 +213,31 @@ def test_big_query_read_new_types(self):
           use_standard_sql=True)))
       assert_that(result, equal_to(expected_data))
 
+  def test_big_query_read_none_values(self):
```
Comment (tvalentyn) [inline sdks/python/apache_beam/io/gcp/bigquery_read_it_test.py]: Please add `@attr('IT')`. I think nose requires it to run filter test scenarios to include into our integration test suites. I am surprised to see that this test actually ran without this decorator... Perhaps because  there is an IT marker on a test with the same prefix `test_bigquery_read`... Anyway, it would be better to either move this decorator to be a class decorator or to add it to every test method.

Comment (tvalentyn) [inline sdks/python/apache_beam/io/gcp/bigquery_read_it_test.py]: Looks like the test failed on Py3.

Comment (tvalentyn) [reply sdks/python/apache_beam/io/gcp/bigquery_read_it_test.py]: Also, how about we incorporate None values to existing IT test scenarios?

**Comment by Juta** [comment]
Run Python PostCommit

**Comment by Juta** [comment]
Run Python PostCommit

**Comment by Juta** [comment]
Run Python PostCommit

**Comment by Juta** [comment]
Run Python PostCommit

**Comment by Juta** [comment]
Run Python PostCommit

**Comment by Juta** [comment]
Run Python PostCommit

**Comment by Juta** [comment]
Run Python PostCommit

**Comment by Juta** [comment]
Run Python PreCommit

**Comment by Juta** [comment]
Run Python PostCommit

**Comment by Juta** [comment]
R: @tvalentyn

File: sdks/python/apache_beam/io/gcp/bigquery_read_it_test.py (hunk 1 above)
Comment (Juta) [reply sdks/python/apache_beam/io/gcp/bigquery_read_it_test.py]: done

**Comment by Juta** [comment]
Run Python PostCommit

**Comment by Juta** [comment]
Run Python PostCommit

**Comment by Juta** [comment]
Thanks @tvalentyn. The PostCommits all succeeded

File: sdks/python/apache_beam/io/gcp/tests/bigquery_matcher.py [hunk 2]
```diff
@@ -144,7 +144,18 @@ def _matches(self, _):
     self.actual_data = response
 
     # Verify result
-    return sorted(self.expected_data) == sorted(self.actual_data)
+    try:
```
Comment (tvalentyn) [inline sdks/python/apache_beam/io/gcp/tests/bigquery_matcher.py]: Is there a reason we cannot reuse https://github.com/apache/beam/blob/2d5e493abf39ee6fc89831bb0b7ec9fee592b9c5/sdks/python/apache_beam/testing/util.py#L117 here?

**Comment by Juta** [comment]
Run Python PostCommit

File: sdks/python/apache_beam/io/gcp/tests/bigquery_matcher.py (hunk 2 above)
Comment (Juta) [reply sdks/python/apache_beam/io/gcp/tests/bigquery_matcher.py]: You are right, it is better to reuse this function. I updated my code

**Comment by Juta** [comment]
Run Portable_Python PreCommit

**Comment by Juta** [comment]
Run Python PreCommit

**Comment by tvalentyn** [comment]
Run Python PostCommit

**Comment by Juta** [comment]
Run Python PostCommit

**Comment by Juta** [comment]
Run Python PostCommit

**Comment by tvalentyn** [comment]
Remaining postcommit failures (test_metrics_it, test_datastore_write_limit)  
are unrelated to the test that is modified in this PR, and failed due to lack of quota (known issue).
@pabloem would you be comfortable merging this PR?

**Comment by tvalentyn** [comment]
Remaining postcommit failures (test_metrics_it, test_datastore_write_limit)
are unrelated to the test that is modified in this PR, and failed due to lack of quota (known issue).
@pabloem would you be comfortable merging this PR?
